### PR TITLE
Add circuitbreaker to lnd

### LINKED ISSUE #228: Add option to run Circuitbreaker for each LND Node
Add the ability to run a custom version of [circuitbreaker](https://github.com/lightningequipment/circuitbreaker) alongside each LND node in warnet. Circuitbreaker can be run using the following command:
`circuitbreaker --network=regtest --rpcserver={lnd host:lnd port} --tlscertpath={path to LND cert} --macaroonpath={path to LND admin.macaroon}`

For my purposes, I'll need to be able to copy in pre-populated databases into the circuitbreaker containers, so it would be very helpful to be able to provide a path to the database that needs to be mounted. This database needs to be in place before circuitbreaker starts (otherwise it'll create its own, empty database), not sure how that works in a kubernetes world.

The database is located at `~/circuitbreaker/circuitbreaker.db` in the using [AppDataDir](https://github.com/btcsuite/btcd/blob/17fdc5219b363c98df12e3ed7849a14f4820d93f/btcutil/appdata.go#L104) to determine the home directory. This base directory can also be customized with `--configdir` if required.

### PR DESCRIPTION
**Pull request by pinheadmz**
Closes: https://github.com/bitcoin-dev-project/warnet/issues/228
Requires (based on) https://github.com/bitcoin-dev-project/warnet/pull/287

Adds one more data element to tanks in the graphml file:

```
 <data key="ln-cb-image">carlakirkcohen/circuitbreaker:endorsement-experiment</data>
```

This adds a container running circuit breaker, and connects it to the corresponding lnd container. In K8s, both containers are in the same pod so they can interact via `localhost` and share a volume at `/root/.lnd` in both containers. In docker they also share a volume with that same path (this is needed to pass circuitbreaker the credentials generated by lnd).

~I'm not sure how else to test circuit breaker~, but the logs look promising. It keeps retrying to connect until it finally does:

```
2024-02-27 12:55:38 2024-02-27T17:55:38.313Z    INFO    Circuit Breaker starting        {"version": ""}
2024-02-27 12:55:38 2024-02-27T17:55:38.313Z    INFO    Opening database        {"path": "/root/.circuitbreaker/circuitbreaker.db"}
2024-02-27 12:55:38 2024-02-27T17:55:38.314Z    ERROR   Unexpected exit {"err": "open /root/.lnd/tls.cert: no such file or directory"}
2024-02-27 12:55:38 main.main
2024-02-27 12:55:38     /src/main.go:144
2024-02-27 12:55:38 runtime.main
2024-02-27 12:55:38     /usr/local/go/src/runtime/proc.go:250
2024-02-27 12:55:44 2024-02-27T17:55:44.878Z    INFO    Circuit Breaker starting        {"version": ""}
2024-02-27 12:55:44 2024-02-27T17:55:44.878Z    INFO    Opening database        {"path": "/root/.circuitbreaker/circuitbreaker.db"}
2024-02-27 12:55:44 2024-02-27T17:55:44.884Z    INFO    Press ctrl-c to exit
2024-02-27 12:55:44 2024-02-27T17:55:44.884Z    INFO    Grpc server starting    {"listenAddress": "127.0.0.1:9234"}
2024-02-27 12:55:44 2024-02-27T17:55:44.884Z    INFO    CircuitBreaker started
2024-02-27 12:55:44 2024-02-27T17:55:44.884Z    INFO    HTTP server starting    {"listenAddress": "127.0.0.1:9235"}
2024-02-27 12:55:44 2024-02-27T17:55:44.886Z    INFO    Connected to lnd node   {"pubkey": "0378b895ecaff7d8a140a1626c46b0d006cd0107dcec2aa261749e338bfc3c17fa"}
2024-02-27 12:55:44 2024-02-27T17:55:44.887Z    INFO    Interceptor/notification handlers registered
```

UPDATE: add rpc `exec_run` to make a request from the circuit breaker local API and check the forwards.

reference:

```
wget -q -O - localhost:9235/api/forwarding_history

{
  "forwards":
  [
    {
      "addTimeNs": "1709062043928206045",
      "resolveTimeNs": "1709062044022878253",
      "settled": true,
      "incomingAmount": "11000",
      "outgoingAmount": "10000",
      "incomingPeer": "000000000000000000000000000000000000000000000000000000000000000000",
      "incomingCircuit":
      {
        "shortChannelId": "123145302441984",
        "htlcIndex": 0
      },
      "outgoingPeer": "000000000000000000000000000000000000000000000000000000000000000000",
      "outgoingCircuit":
      {
        "shortChannelId": "123145302376448",
        "htlcIndex": 0
      },
      "incomingEndorsed": false,
      "outgoingEndorsed": false,
      "cltvDelta": 40
    }
  ]
}
```

### CODE REVIEW

**Comment by pinheadmz** [comment]
TODO: Still gotta figure out how to get pre-populated databases into those new cb containers

**Comment by pinheadmz** [comment]
rebased on main after #290 so test base healthcheck should wait for circuit breakers, which was causing failures before

File: src/backends/compose/compose_backend.py [hunk 1]
```diff
@@ -92,15 +92,15 @@ def up(self, *_):
             )
 
     def down(self, warnet):
-        command = ["docker", "compose", "down"]
+        command = ["docker", "compose", "down", "-v"]
```
Comment (willcl-ark) [inline src/backends/compose/compose_backend.py:96]: Nice tidy-up.

File: src/backends/kubernetes/kubernetes_backend.py [hunk 2]
```diff
@@ -609,11 +632,30 @@ def deploy_pods(self, warnet):
...
+                if tank.lnnode.cb:
+                    # Create a shared volume between containers in the pod
+                    volume_name = f"ln-cb-data-{tank.index}"
+                    vols.append(client.V1Volume(
+                        name=volume_name,
+                        empty_dir=client.V1EmptyDirVolumeSource()
+                    ))
+                    volume_mounts.append(client.V1VolumeMount(
+                        name=volume_name,
+                        mount_path='/root/.lnd'
```
Comment (willcl-ark) [inline src/backends/kubernetes/kubernetes_backend.py]: As this is also used in `create_circuitbreaker_container` IMO it makes sense to extract it as a variable at the head of the file?

**Review by willcl-ark — APPROVED** [review]
Sweet looks good to me. Ran the test locally which passed. left a little feedback you could choose to address or ignore.

Offtopic: I am going to clean up the logging, as it's become annoying to parse (debug statements added by me!) and review new changes.

**Pull request merged**